### PR TITLE
social studies subject page

### PR DESCRIPTION
**Pull request by emilia-friedberg**
## WHAT
Build out the social studies subject page.

## WHY
This completes the flow for assigning social studies content, which will be demo-ed to the OER Project partners on Wednesday.

## HOW
Mostly just straightforward HTML and CSS, with some wiring in the backend to make this page accessible from the various routes. This page does use slightly redesigned versions of the Q&A sections and blog post preview cards that appear on other pages, so I made some updates there to consolidate the CSS for that code in order to apply the updates everywhere.

### Screenshots
Screenshot 2024-08-05 at 10 40 30 AM
Screenshot 2024-08-05 at 10 40 37 AM
Screenshot 2024-08-05 at 10 41 12 AM
Screenshot 2024-08-05 at 10 41 21 AM
Screenshot 2024-08-05 at 10 41 31 AM
Screenshot 2024-08-05 at 10 41 38 AM

### Notion Card Links
https://www.notion.so/quill/Subject-Page-for-Social-Studies-Interdisciplinary-Science-Offering-779253ff3e92474fb6d3ddf04d236f30?pvs=4

### What have you done to QA this feature?
Tested on staging both logged-in and logged-out, and also made sure the page works on mobile. Jack and Jamie are QAing as well.

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? | YES
Have you deployed to Staging? | YES
Self-Review: Have you done an initial self-review of the code below on Github? | YES
Screenshot 2024-08-05 at 10 41 52 AM

### CODE REVIEW

File: services/QuillLMS/app/assets/stylesheets/pages/blog_posts.scss [hunk 1]
```diff
@@ -64,13 +64,13 @@ $blogpostlightteal: rgba(44, 127, 155, 0.08);
...
-    .quill-button-archived {
+    .quill-button {
       background-color: $blogpostpurple;
     }
-    .quill-button-archived:hover, .quill-button-archived:focus {
+    .quill-button:hover, .quill-button:focus {
       background-color: $blogpostpurple !important;
     }
-    .quill-button-archived:focus {
+    .quill-button:focus {
```
Comment (eadams17) [inline services/QuillLMS/app/assets/stylesheets/pages/blog_posts.scss:73]: I'm glad we went this route-- will make these updates much easier!

File: services/QuillLMS/client/app/bundles/Teacher/containers/SocialStudiesSubjectPage.tsx [hunk 2]
```diff
@@ -0,0 +1,196 @@
+import * as React from 'react';
+import ReactMarkdown from 'react-markdown';
+
+import QuestionsAndAnswers from './QuestionsAndAnswers'
+
+import PreviewCard from '../components/shared/preview_card.jsx';
+import { requestGet } from '../../../modules/request';
+
+const largeIconBaseSrc = `${process.env.CDN_URL}/images/icons/l`
```
Comment (eadams17) [inline services/QuillLMS/client/app/bundles/Teacher/containers/SocialStudiesSubjectPage.tsx]: maybe we should add this to the Shared bundle and make this a pattern with other icon sizes?

**Review by eadams17 — APPROVED** [review]
LGTM!

**Review by anathomical — APPROVED** [review]

File: services/QuillLMS/client/app/bundles/Teacher/containers/SocialStudiesSubjectPage.tsx (hunk 2 above)
Comment (emilia-friedberg) [reply services/QuillLMS/client/app/bundles/Teacher/containers/SocialStudiesSubjectPage.tsx]: Good idea!

**Pull request merged**